lib/api/core/plugins/pluginsManager.js

```diff
@@ -714,130 +708,87 @@ class PluginsManager {
-    pluginsDir = path.resolve(path.join(rootPath, 'plugins/enabled')),
-    loadedPlugins = {};
-
-  let pluginList = [];
-
-  try {
-    pluginList = fs.readdirSync(pluginsDir);
-  }
-  catch(e) {
-    throw new PluginImplementationError(`Unable to load plugins from directory "${pluginsDir}"; ${e.message}`);
-  }
-
-  pluginList = pluginList.filter(element => {
-    const pluginPath = path.join(pluginsDir, element);
-
-    try {
-      return fs.statSync(pluginPath).isDirectory();
-    }
-    catch(e) {
-      throw new PluginImplementationError(`Unable to load plugin from path "${pluginPath}"; ${e.message}`);
-    }
-  });
-
-  debug('loading plugins: %a', pluginList);
-
-  pluginList.forEach(pluginDirName => {
-    let packageJson;
-    const pluginPath = path.resolve(pluginsDir, pluginDirName);
+  /**
+   * Load detected plugins in memory
+   *
+   * @returns {object} list of loaded plugin
+   */
+  load() {
+    const loadedPlugins = {};
 
+    let plugins = [];
     try {
-      packageJson = require(path.resolve(pluginPath, 'package.json'));
+      plugins = fs.readdirSync(this.pluginsDir);
     }
     catch(e) {
-      throw new PluginImplementationError(`Unable to load plugin from path "${pluginDirName}"; No package.json found.`);
-    }
-
-    if (!packageJson.name || packageJson.name.length === 0) {
-      throw new PluginImplementationError(`Unable to load plugin from path "${pluginDirName}"; No "name" property provided in package.json`);
+      throw new PluginImplementationError(`Unable to load plugins from directory "${this.pluginsDir}"; ${e.message}`);
     }
 
-    // We need to ignore the case of plugin names while checking for name conflicts,
-    // because we have to lowercase the name of the plugin to create a dedicated
-    // Elasticsearch index.
-    const lowercased = packageJson.name.toLowerCase();
+    for (const plugin of plugins) {
+      const pluginPath = path.join(this.pluginsDir, plugin);
 
-    for (const name of Object.keys(loadedPlugins)) {
-      if (lowercased === name.toLowerCase()) {
-        throw new PluginImplementationError(`A plugin named ${lowercased} already exists`);
+      try {
+        fs.statSync(pluginPath).isDirectory();
+      }
+      catch(e) {
+        throw new PluginImplementationError(`Unable to load plugin from path "${pluginPath}"; ${e.message}`);
       }
     }
 
-    const
-      plugin = {
-        name: packageJson.name,
-        path: pluginPath,
-        object: null,
-        config: {},
-        manifest: Object.assign({}, defaultManifest)
-      };
-
-    debug('[%s] loading plugin from directory "%s"', plugin.name, plugin.path);
+    debug('loading plugins: %a', plugins);
 
-    try {
-      plugin.manifest = Object.assign(plugin.manifest, require(path.resolve(plugin.path, 'manifest.json')));
-    }
-    catch(e) {
-      // do nothing, simply use default manifest
-    }
+    for (let pluginPath of plugins) {
+      pluginPath = path.resolve(this.pluginsDir, pluginPath);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove this assignment of "pluginPath". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS2310)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the foreach variable "pluginPath". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1226)